File: Data Structures & Algorithms/top-k-elements-in-list/submission-13.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        buckets = [[] for _ in range(len(nums)+1)]

        freq = {}

        for num in nums:
            freq[num] = freq.get(num,0) + 1
        
        for num, count in freq.items():
            buckets[count].append(num)
        
        result = []

        for i in range(len(buckets)-1, -1, -1):

            if len(buckets[i]) == 0:
                continue
            
            
            for n in buckets[i]:
                result.append(n)
                if len(result) == k:
                    return result
        return result

File: Data Structures & Algorithms/is-anagram/submission-13.java
class Solution {
    public boolean isAnagram(String s, String t) {

        if(s.length() != t.length()){
            return false;
        }

        int[] freq = new int[26];

        for(char ch : s.toCharArray()){
            freq[ch - 'a'] += 1;
        }

        for(char ch : t.toCharArray()){
            freq[ch - 'a'] -= 1;
        }

        for(int i : freq){
            if (i!=0) return false;
        }

        return true;

    }
}
